src/project/SendButton.tsx

```diff
@@ -0,0 +1,100 @@
+/**
+ * (c) 2021, Micro:bit Educational Foundation and contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+import {
+  Button,
+  ButtonGroup,
+  HStack,
+  IconButton,
+  Menu,
+  MenuButton,
+  MenuItem,
+  MenuList,
+  Portal,
+  ThemeTypings,
+  Tooltip,
+} from "@chakra-ui/react";
+import React, { useCallback } from "react";
+import { MdMoreVert } from "react-icons/md";
+import { RiUsbLine } from "react-icons/ri";
+import { FormattedMessage, useIntl } from "react-intl";
+import { zIndexAboveTerminal } from "../common/zIndex";
+import { ConnectionStatus } from "../device/device";
+import { useConnectionStatus } from "../device/device-hooks";
+import { useProjectActions } from "./project-hooks";
+
+interface SendButtonProps {
+  size?: ThemeTypings["components"]["Button"]["sizes"];
+}
+
+const SendButton = ({ size }: SendButtonProps) => {
+  const status = useConnectionStatus();
+  const connected = status === ConnectionStatus.CONNECTED;
+  const actions = useProjectActions();
+  const handleToggleConnected = useCallback(async () => {
+    if (connected) {
+      await actions.disconnect();
+    } else {
+      await actions.connect();
+    }
+  }, [connected, actions]);
+  const handleSendToMicrobit = useCallback(async () => {
+    await actions.flash();
+  }, [actions]);
```

Comment:
This can probably just be replaced with passing actions.flash to the onClick.

Reply:
[`2cf98b7`](https://github.com/microbit-foundation/python-editor-next/pull/763/commits/2cf98b77f736e0600c02f978367491783250b2f4).